tools-tests/parser_test.py

```diff
@@ -11,3 +11,14 @@
 
 def test_parser():
     sgcollect_info.create_option_parser()
+
+
+def test_parser_log_redaction_salt():
+    parser = sgcollect_info.create_option_parser()
+    options, _ = parser.parse_args(["--log-redaction-salt=a", "foo.zip"])
+    assert options.salt_value == b"a"
+
+    options, _ = parser.parse_args(["foo.zip"])
+    # assert this is bytes uuid4
+    assert isinstance(options.salt_value, bytes)
+    assert len(options.salt_value) == 16
```

Comment:
I think this may be an incorrect assertion / change. See comparions against `main` here:

![Screenshot 2025-02-21 at 17 12 13](https://github.com/user-attachments/assets/38ed1ade-d893-4903-af73-04e02d5a8739)

![Screenshot 2025-02-21 at 17 32 32](https://github.com/user-attachments/assets/ad209d0f-3030-4328-a132-9357e0f81eaa)


Seems like we would want to preserve the string version of `salt_value`, or at least make it a stringified UUID stored as bytes.

The hash is consistent if explicitly given a salt, so perhaps we can just change the `default=` initialization to correct this and accept `b'salt_value'` in the unredacted output.

Reply:
Really good catch, fixed.